dev/package.json

```diff
@@ -27,18 +27,32 @@
         "prepack": "tsc -p ./",
         "compile": "tsc -watch -p ./",
         "lint": "tslint --project tsconfig.json -e src/*.d.ts -t verbose",
-        "test": "mocha out/test/**/*.js --ui tdd --reporter mocha-junit-reporter",
+        "test": "mocha out/test/**/*.js --ui tdd --reporter mocha-multi-reporters --reporter-options configFile=reporter-options.json",
         "prepare": "node ./node_modules/vscode/bin/install"
     },
     "dependencies": {},
     "devDependencies": {
+        "@types/clean-webpack-plugin": "^0.1.2",
+        "@types/copy-webpack-plugin": "^4.4.2",
```

Comment:
I still think almost all of these should be `dependencies` instead of `devDependencies`. That could be why you needed to turn off `no-implicit-dependencies`, which is a rule I actually really like. I get that it's counterintuitive, but this is already a dev package and it relies on these packages for it's main purpose. It won't be a problem in the extensions as long as they list `vscode-azureextensiondev` as _their_ devDependency

Reply:
I remember we talked about this before but couldn't remember what we'd decided, but everything right now was dev dependencies so I kept it that way (was thinking you had argued for that, couldn't remember why, maybe not).  We can either keep them dev dependencies and force users to also add as dev dependencies, or do them as regular dependencies and add the third party notices work (even though the actual using product won't ship with them). Don't have a strong opinion. Would it be easier for me to do a third party notice manually (have to do before publishing), or for you to set it up in the pipeline?

Reply:
> everything right now was dev dependencies so I kept it that way

Well yes you've added everything so far :P  (except maybe one)

> was thinking you had argued for that

Nope https://github.com/Microsoft/vscode-azuretools/pull/277#discussion_r222777914. I argued _for_ listing them in dependencies and you said "I think I'll leave it as is for now". I think it's wrong to list these as devDependencies and feel more strongly the more you add. It results in duplicate dependencies listed in the extensions and IMO leads to more bugs. For example if you remove or add a devDependency here that means you have to remember to do the same in every single extension repo. You're losing the whole benefit of npm's dependency structure that would handle this for us.

> Would it be easier for me to do a third party notice manually (have to do before publishing), or for you to set it up in the pipeline?

Easier for you to do one manually. I haven't done a whole lot of TPN stuff in this repo since we don't plan to webpack the packages directly.

Reply:
Moved to dev deps.